packages/front-end/pages/segments/index.tsx

```diff
@@ -333,13 +334,25 @@ const SegmentPage: FC = () => {
           </div>
         </div>
       )}
-      {!hasFileConfig() && segments.length === 0 && (
+      {segments.length === 0 && !hasFileConfig() && (
         <div className="alert alert-info">
           You don&apos;t have any segments defined yet.{" "}
           {permissions.createSegments &&
             "Click the button above to create your first one."}
         </div>
       )}
+      {segments.length === 0 &&
+        hasFileConfig() &&
+        !process.env.STORE_SEGMENTS_IN_MONGO && (
```

Comment:
You can't use `process.env` on the front-end like this. Environment variables in the front-end are only evaluated once at build time (in our CI/CD pipeline) and then hard-coded in the resulting build.

Instead, look at `front-end/pages/api/init.ts` and `front-end/services/env.ts`.  To get environment variables in the front-end, we make an API call when the app first loads and store the result in memory.

Reply:
Thank you! I had a feeling it might be a bit wonky. Just updated.